File: frontend/src/backend/shoppingService.js
import {appFetch} from './appFetch';

export const addToShoppingCart = async (shoppingCartId, productId, quantity) =>
    await appFetch('POST', `/shopping/shoppingcarts/${shoppingCartId}/addToShoppingCart`, 
        {productId, quantity});

export const updateShoppingCartItemQuantity = async (shoppingCartId, productId, quantity) =>
    await appFetch('POST', `/shopping/shoppingcarts/${shoppingCartId}/updateShoppingCartItemQuantity`, 
        {productId, quantity});

export const removeShoppingCartItem = async (shoppingCartId, productId) =>
    await appFetch('POST', `/shopping/shoppingcarts/${shoppingCartId}/removeShoppingCartItem`, 
        {productId});

export const buy = async (shoppingCartId, postalAddress, postalCode) =>
    await appFetch('POST', `/shopping/shoppingcarts/${shoppingCartId}/buy`, 
        {postalAddress, postalCode});

export const findOrders = async ({page}) => 
    await appFetch('GET', `/shopping/orders?page=${page}`);

export const findOrder = async orderId =>
    await appFetch('GET', `/shopping/orders/${orderId}`);

File: backend/src/main/java/es/udc/pashop/backend/model/services/ShoppingService.java
package es.udc.pashop.backend.model.services;

import es.udc.pashop.backend.model.exceptions.*;
import es.udc.pashop.backend.model.entities.Order;
import es.udc.pashop.backend.model.entities.ShoppingCart;

public interface ShoppingService {
	
	ShoppingCart addToShoppingCart(Long userId, Long shoppingCartId, Long productId, int quantity) 
		throws InstanceNotFoundException, PermissionException, MaxQuantityExceededException, MaxItemsExceededException;
	
	ShoppingCart updateShoppingCartItemQuantity(Long userId, Long shoppingCartId, Long productId, int quantity)
		throws InstanceNotFoundException, PermissionException, MaxQuantityExceededException;
	
	ShoppingCart removeShoppingCartItem(Long userId, Long shoppingCartId, Long productId)
		throws InstanceNotFoundException, PermissionException;
	
	Order buy(Long userId, Long shoppingCartId, String postalAddress, String postalCode)
		throws InstanceNotFoundException, PermissionException, EmptyShoppingCartException;
	
	Order findOrder(Long userId, Long orderId) throws InstanceNotFoundException, PermissionException;
	
	Block<Order> findOrders(Long userId, int page, int size);

}
